src/activations.jl

```diff
@@ -785,58 +840,84 @@ this replacement for some array or element types.
 
 ## This is a performance hack specifically for Zygote, because it doesn't handle fused
 ## broadcasts well; but it generally should be good (or at least harmless) for any AD, as
 ## it saves ADing the broadcasting machinery.
 ## Related Issue https://github.com/JuliaDiff/ChainRulesCore.jl/issues/271
 
-UNARY_ACTS = [ # f, df
-    (:relu,         :(x > 0)),
-    (:hardtanh,     :(-1 < x < 1)),
-    (:selu,         :(deriv_selu(Ω))),
+## TODO: add to the lists below all activations.
+
+UNARY_ACTS = [ # f, dfdx
+    ## In the same order as above!
     (:σ,            :(conj(Ω * (1 - Ω)))),
+    (:hardσ,        :(ifelse((Ω>0)&(Ω<1), 1//6, 1//1))),
+    (:logσ,         :(sigmoid_fast(-x))),
+    (:hardtanh,     :((Ω>-1) & (Ω<1))),
+    (:relu,         :(Ω > 0)),
+    (:leakyrelu,    :(ifelse(Ω > 0, 1//1, 1//100))),
+    (:relu6,        :((Ω>0) & (Ω<6))),
+    # rrelu is random, can't write a rule.
     (:elu,          :(deriv_elu(Ω))),
-    (:softplus,     :(σ(x))),
-
+    # gelu
+    (:swish,        :(Ω + sigmoid_fast(x) * (1 - Ω))),
+    (:hardswish,    :(deriv_hardswish(x))),
+    # lisht
+    (:selu,         :(deriv_selu(Ω))),
+    (:celu,         :(deriv_celu(Ω))),
+    (:trelu,        :(Ω > 0)),
+    (:softsign,     :(deriv_softsign(x))),
+    (:softplus,     :(sigmoid_fast(x))),
+    # (:softplus,     :(1 - @fastmath exp(-Ω))),  # slightly faster, check accuracy?
+    # logcosh
+    # mish
+    (:tanhshrink,    :((x - Ω)^2)),
+    (:softshrink,    :(Ω != 0)),
+    ## Fast variants are the same!
     (:tanh_fast,    :(conj(1 - Ω^2))),
     (:sigmoid_fast, :(conj(Ω * (1 - Ω)))),
-    ]
+]
 
-for (f, df) in UNARY_ACTS
-    @eval @scalar_rule($f(x), $df)
+for (f, dfdx) in UNARY_ACTS
+    @eval @scalar_rule($f(x), $dfdx)
 
     pullback = Symbol(:broadcasted_, f, :_pullback)
     @eval function rrule(::typeof(broadcasted),
                          ::typeof($f), x::Numeric)
         Ω = $f.(x)
-        function $pullback(Δ)
+        function $pullback(dΩ)
             x_thunk = InplaceableThunk(
-                dx -> @.(dx += Δ * $df),
-                @thunk @.(Δ * $df)
+                dx -> @.(dx += dΩ * $dfdx),
+                @thunk @.(dΩ * $dfdx)
             )
             NoTangent(), NoTangent(), x_thunk
         end
         return Ω, $pullback
     end
 end
 
-BINARY_ACTS = [ # f, df1, df2
-    (:elu, :(deriv_elu(Ω, x2)), :(NoTangent())), # TODO use real deriv instead of DNE
-    ]
+# NO_ACT_GRAD = ChainRulesCore.@not_implemented "for simplicitly NNlib assumes the 2nd argument of this activation function is a constant"
+NO_ACT_GRAD = NaN  ## Still reminds you not to use this, but is perhaps more GPU friendly.
 
-for (f, df1, df2) in BINARY_ACTS
-    @eval @scalar_rule($f(x1, x2), ($df1, $df2))
+BINARY_ACTS = [ # f, dfdx1, dfdx2
+    ## In the same order as above!
+    (:leakyrelu,   :(ifelse(Ω > 0, oftf(Ω, 1), oftf(Ω, x2))), NO_ACT_GRAD),
+    (:elu,         :(deriv_elu(Ω, x2)),      NO_ACT_GRAD),
```

Comment:
What rules like this don't solve is that you need to change `Dense(10 => 20, lekyrelu)` into
```
Chain(Dense(10 => 20), x -> leakyrelu.(x, 0.1))
```
to provide the 2nd argument. 

Not this PR, but it would be nice if you could write `Dense(10 => 20, lekyrelu(0.1))` instead. Perhaps it has to be `Dense(10 => 20, Lekyrelu(0.1))` with some new struct, but perhaps you can just be clever with overloading `broadcasted` so that `lekyrelu(0.1)` makes a function but `Dense(10 => 20, lekyrelu)` doesn't create a vector of functions?